`Could not find a declaration file for module '../src/runtime/control.js'` error when typechecking `svelte.config.js` with `noImplicitAny`

### Describe the bug

by default, type checking seems to be disabled on `svelte.config.js`, despite the fact that it was generated with `@type` jsdocs

```js
import adapter from '@sveltejs/adapter-auto';

/** @type {import('@sveltejs/kit').Config} */
const config = {
	kit: {
		adapter: adapter()
	}
};

export default config;
```

however, attempting to type check it causes the following error:
```
❯ tsc --allowjs --checkjs --noemit --module esnext --noimplicitany --moduleresolution node .\svelte.config.js
node_modules/@sveltejs/kit/types/index.d.ts:20:37 - error TS7016: Could not find a declaration file for module '../src/runtime/control.js'. '/home/projects/sveltejs-kit-template-default-q26evm/node_modules/@sveltejs/kit/src/runtime/control.js' implicitly has an 'any' type.

20 import { HttpError, Redirect } from '../src/runtime/control.js';
                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in node_modules/@sveltejs/kit/types/index.d.ts:20
```
### Reproduction

https://stackblitz.com/edit/sveltejs-kit-template-default-q26evm?file=svelte.config.js&terminal=dev

### Logs

_No response_

### System Info

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (12) x64 11th Gen Intel(R) Core(TM) i5-11500T @ 1.50GHz
    Memory: 1.52 GB / 15.73 GB
  Binaries:
    Node: 18.10.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.19.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (103.0.1264.62)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.81
    @sveltejs/kit: next => 1.0.0-next.512
    svelte: ^3.44.0 => 3.51.0
    vite: ^3.1.0 => 3.1.7
```


### Severity

annoyance

### Additional Information

_No response_

--- a/.changeset/lucky-gifts-jump.md
+++ b/.changeset/lucky-gifts-jump.md
@@ -1,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+[fix] export HttpError/Redirect interface

--- a/packages/kit/types/index.d.ts
+++ b/packages/kit/types/index.d.ts
@@ -18,7 +18,6 @@ import {
 	UniqueInterface
 } from './private.js';
 import { SSRNodeLoader, SSRRoute, ValidatedConfig } from './internal.js';
-import { HttpError, Redirect } from '../src/runtime/control.js';
 
 export { PrerenderOption } from './private.js';
 
@@ -52,13 +51,6 @@ type OptionalUnion<
 	U extends Record<string, any>, // not unknown, else interfaces don't satisfy this constraint
 	A extends keyof U = U extends U ? keyof U : never
 > = U extends unknown ? { [P in Exclude<A, keyof U>]?: never } & U : never;
-
-// Needs to be here, else ActionData will be resolved to unknown - probably because of "d.ts file imports .js file" in combination with allowJs
-export interface ValidationError<T extends Record<string, unknown> | undefined = undefined>
-	extends UniqueInterface {
-	status: number;
-	data: T;
-}
 
 type UnpackValidationError<T> = T extends ValidationError<infer X>
 	? X
@@ -753,6 +745,16 @@ export function error(
 ): HttpError;
 
 /**
+ * The object returned by the `error` function
+ */
+export interface HttpError extends UniqueInterface {
+	/** The HTTP status code */
+	status: number;
+	/** The error message */
+	body: { message: string } extends App.Error ? App.Error | string | undefined : App.Error;
+}
+
+/**
  * Creates a `Redirect` object. If thrown during request handling, SvelteKit will
  * return a redirect response.
  */
@@ -760,6 +762,16 @@ export function redirect(
 	status: 300 | 301 | 302 | 303 | 304 | 305 | 306 | 307 | 308,
 	location: string
 ): Redirect;
+
+/**
+ * The object returned by the `redirect` function
+ */
+export interface Redirect extends UniqueInterface {
+	/** The HTTP status code */
+	status: 300 | 301 | 302 | 303 | 304 | 305 | 306 | 307 | 308;
+	/** The location to redirect to */
+	location: string;
+}
 
 /**
  * Generates a JSON `Response` object from the supplied data.
@@ -773,3 +785,12 @@ export function invalid<T extends Record<string, unknown> | undefined>(
 	status: number,
 	data?: T
 ): ValidationError<T>;
+
+/**
+ * The object returned by the `invalid` function
+ */
+export interface ValidationError<T extends Record<string, unknown> | undefined = undefined>
+	extends UniqueInterface {
+	status: number;
+	data: T;
+}

